### ISSUE TITLE
"matches()" doesn't use aliases

### ISSUE THREAD

**Issue by Proliner** [alexgarrett/violin#90]

**Comment by pedzed** [comment]
What do you exactly mean? Could you add a description?

**Comment by ddouglas493** [comment]
I can elaborate on this since i have encountered a similar issue.

When using the matches rule, the system delivers the error message using the field name rather than than the alias

`$v->validate([`
`         'new_email|New Email Address' => [Input::get('new_email'), "required|email"],`
`        'new_email_confirm|New Email Address Again' => [Input::get('new_email_confirm'), "required|email|matches()"]`
`      ]);`

Returns

> New Email Address Again must match 'new_email'.

This rule should be able to detect when an alias is in use and implement the alias rather than the field name

**Comment by pedzed** [comment]
Oh, thanks for the clarification @ddouglas493. Now I get it. It would be nice if you could create a PR for this. Otherwise I might do that later if I have some spare time.

**Comment by ddouglas493** [comment]
Yeah, I am really not sure what needs to be done to fix the issue so I cannot submit the PR. I have looked over the code and attempted but failed to fix it myself.

**Comment by dignityinside** [comment]
Some one fix this already?
